app/src/org/commcare/network/HttpRequestEndpointsMock.java

```diff
@@ -31,13 +31,21 @@ public static void setCaseFetchResponseCodes(Integer[] responseCodes) {
 
     @Override
     public HttpResponse makeCaseFetchRequest(String baseUri, boolean includeStateFlags) throws ClientProtocolException, IOException {
-        int responseCode;
-        if (caseFetchResponseCodeStack.size() > 0) {
-            responseCode = caseFetchResponseCodeStack.remove(0);
+        if (LocalDataPullResponseFactory.INSTANCE.isAsyncRestore) {
+            return makeAsyncCaseFetchRequest();
```

Comment:
Since `makeAsyncCaseFetchRequest` is only called here and is one line, maybe you could just inline its body here?